### ISSUE TITLE
Unified instance directory (Shared Folders)

### ISSUE THREAD

**Issue by ExecuteOrd66** [gorilla-devs/GDLauncher#994]
**Is your feature request related to a problem? Please describe.**
1) Multiple copies of resource-packs, mods and world-saves across instances waste storage space on the computer. 
2) When changes are made to a resource-pack, mod or world-save (which the user would like to have across multiple instances) the user would have to manually copy the latest version to all the independent instances.

**Describe the solution you'd like**
Scenario: Have multiple 1.16.5 mod-packs installed. These mod-packs have many overlapping mods. 
Possible solution: An optional unified folder that pools resource-packs, mods and world-saves that can be loaded independently by each instance. 1.16.5 mods are stored under a folder named "lib\mods" (just as an example) and a list of the required mods for each mod-pack is stored separately. The modloader is redirected to the "lib\mods" directory and only loads the instance's corresponding mod files.

**Describe alternatives you've considered**
An optional unified folder pools resource-packs, mods and world-saves. When an instance is launched, all files in the aforementioned folder is synced to the corresponding instance, to ensure the latest version of each file is present. If changes are made to a world-save in the instance, changes are also synced to the pooled folder (This would only address the 2nd problem, but is simple to implement compared to the first possible solution).

**Comment by ExecuteOrd66** [comment]
Actually found out about symlinks, which is exactly what I needed, but I'll keep the thread open if the devs are still interested. This would unfortunately not work with mods though

**Comment by Mari023** [comment]
would be a cool feature to have all mods/texturepacks/shaderpacks/datapacks stored centrally and just symlinked.
although the storage saving is probably negligible for most users.

**Comment by ExecuteOrd66** [comment]
I poked around the settings and saw "Clear Shared Data". Mabey it already exists?

**Comment by Mari023** [comment]
I think only for minecraft/fabric/forge versions

**Comment by blarfoon** [comment]
This is something we considered but I’m not entirely sure if it’s worth

**Comment by YoYoK3nny** [comment]
Not everyone would need it, but it'll definitely be worth it for those who have several modpacks that all use identical mods/resourcepacks.

It'll also make installing modpacks faster since the same mods won't need to be downloaded multiple times (especially helpful for large mods) and creating links is faster than copying files.

**Comment by XaaRii** [comment]
I personally would love to have one shared server list (and also resource packs, shaders, etc. like mentioned above)
I'm thinking of making GDLauncher my primary launcher, as i always play on my server with a bunch of clientside mods. Update feature is great and all, but the lack of defining shared files (or creating a default template, idk) makes me wanna reconsider.
Not having a shared server list (or any settings on this note) feels like a no-brainer to me, but hey, that's just my opinion. Maybe it's just me, since i haven't seen anyone else crying about it (or maybe just everyone loves singleplayer and lacks friends, i dunno)

**Comment by Eskaan** [comment]
I don't think this will happen in this major version of GDLauncher, symlinks also bring many problems, for example manual mod editing gets nearly impossible, it needs detection for new mods to then move them and so on.
